pkg/cmd/server/server.go

```diff
@@ -262,28 +270,43 @@ func newServer(f client.Factory, config serverConfig, logger *logrus.Logger) (*s
 		return nil, err
 	}
 
+	// cancelFunc is not deferred here because if it was, then ctx would immediately
+	// be cancelled once this function exited, making it useless to any informers using later.
+	// That, in turn, causes the velero server to halt when the first informer tries to use it (probably restic's).
+	// Therefore, we must explicitly call it on the error paths in this function.
 	ctx, cancelFunc := context.WithCancel(context.Background())
 
 	clientConfig, err := f.ClientConfig()
 	if err != nil {
+		cancelFunc()
 		return nil, err
 	}
 
+	var csiSnapClient *snapshotvebeta1client.Clientset
+	if features.IsEnabled("EnableCSI") {
```

Comment:
suggest creating constants for feature names and export them. That way, we can use the same constant in the plugin as well to decide whether or not to register the plugins

Reply:
I like the idea of making the string a constant & exporting it, but I'm not sure about registering the plugin based on it. That decision can be done outside this PR, though.